src/Erc20.ts

```diff
@@ -12,49 +12,106 @@ const contractABI = require("../abis/ERC20.json");
 export async function getErc20TokenDetails(
   contractAddress: string,
   chainId: number
-): Promise<{ readonly name: string; readonly decimals: number; readonly symbol: string }> {
+): Promise<{
+  readonly name: string;
+  readonly decimals: number;
+  readonly symbol: string;
+}> {
+  console.log(
+    `[getErc20TokenDetails] Starting for address: ${contractAddress}, chainId: ${chainId}`
+  );
+
   const cache = Cache.init(CacheCategory.Token, chainId);
   const token = cache.read(contractAddress.toLowerCase());
 
   if (token) {
+    console.log(
+      `[getErc20TokenDetails] Cache hit for address: ${contractAddress}`
+    );
     return {
       decimals: Number(token.decimals),
       name: token.name,
       symbol: token.symbol,
     };
   }
 
-  // RPC URL
+  console.log(
+    `[getErc20TokenDetails] Cache miss for address: ${contractAddress}`
+  );
+
   const rpcURL = CHAIN_CONSTANTS[chainId].rpcURL;
+  console.log(`[getErc20TokenDetails] Using RPC URL: ${rpcURL}`);
 
-  // Create Web3 instance
   const web3 = new Web3(rpcURL);
-
-  // Create ERC20 contract instance
   const erc20token = new web3.eth.Contract(contractABI, contractAddress);
 
   try {
-    // Use Promise.all to execute all calls in parallel and wait for all of them to resolve
+    console.log(
+      `[getErc20TokenDetails] Fetching token details for address: ${contractAddress}`
+    );
+
     const [name, decimals, symbol] = await Promise.all([
-      erc20token.methods.name().call(),
-      erc20token.methods.decimals().call(),
-      erc20token.methods.symbol().call(),
+      erc20token.methods
+        .name()
+        .call()
+        .catch((e) => {
+          console.error(
+            `[getErc20TokenDetails] Error fetching name: ${e.message}`
+          );
+          return "";
```

Comment:
I believe some meme coin decided to break the ERC20 status quo and do something custom and dogey with on e of these methods and that probably broke things